cpu/stm32f4/periph/spi.c

```diff
@@ -102,6 +111,9 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
             return -2;
     }
 
+    /* initialize lock */
+    mutex_init(&spi_config[dev].lock);
```

Comment:
Does mutex_init() handle multiple initializations? It would be useful to prevent multiple calls to spi_init from trampling each other.
On the other hand, I would prefer that all calls to `spi_init` would be moved to each platform's `board.c` instead of letting the device driver reinitialize the SPI bus after another device driver has already began using it. For example: at86rf231 contains a call to `spi_init` inside its init function.